src/components/discovery/DiscoverySearchCondition.js

```diff
@@ -34,217 +48,164 @@ export const getSchemaTypeTransformer = type => {
     }
 };
 
+const DEFAULT_FIELD_SCHEMA = {
+    search: {...DEFAULT_SEARCH_PARAMETERS},
+};
 
-class DiscoverySearchCondition extends Component {
-    static getDerivedStateFromProps(nextProps) {
-        return "value" in nextProps
-            ? {...(nextProps.value ?? {})}
-            : null;
-    }
-
-    constructor(props) {
-        super(props);
-
-        const value = this.props.value ?? {};
-        this.state = {
-            field: value.field ?? undefined,
-            field2: value.field2 ?? undefined,
-
-            fieldSchema: value.fieldSchema ?? {
-                search: {...DEFAULT_SEARCH_PARAMETERS},
-            },
-            fieldSchema2: value.fieldSchema2 ?? {
-                search: {...DEFAULT_SEARCH_PARAMETERS},
-            },
-
-            negated: value.negated ?? false,
-            operation: value.operation ?? OP_EQUALS,
-            searchValue: value.searchValue ?? "",
+const DiscoverySearchCondition = ({ dataType, value, onChange, onFieldChange, isExcluded, onRemoveClick }) => {
+    const [fieldState, setFieldState] = useState({
+        field: value.field ?? undefined,
+        fieldSchema: value.fieldSchema ?? DEFAULT_FIELD_SCHEMA,
+        negated: value.negated ?? false,
+        operation: value.operation ?? OP_EQUALS,
+        searchValue: value.searchValue ?? "",
+    });
+
+    useEffect(() => {
+        if (value) {
+            setFieldState({
+                field: value.field ?? undefined,
+                fieldSchema: value.fieldSchema ?? DEFAULT_FIELD_SCHEMA,
+                negated: value.negated ?? false,
+                operation: value.operation ?? OP_EQUALS,
+                searchValue: value.searchValue ?? "",
+            });
+        }
+    }, [value]);
 
-            // Non-value props
-            dataType: props.dataType ?? null,
-            dataType2: props.dataType2 ?? null,
-        };
+    const handleChange = useCallback((change) => {
+        const newState = {...fieldState, ...change};
+        if (value === undefined) setFieldState(newState);
+        if (onChange) onChange(newState);
+    }, [fieldState, onChange]);
 
-        this.handleChange = this.handleChange.bind(this);
-    }
+    const { field, fieldSchema, operation, negated, searchValue } = fieldState;
 
-    handleField = (value, key = "field", fieldSchemaKey = "fieldSchema") => {
-        if (this.state[key] === value.selected) return;
+    const handleField = useCallback((value) => {
```

Comment:
I would use a different name for the value callback argument, since "value" is one of the component's props too